substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/NativeLibrarySupport.java

```diff
@@ -120,22 +120,29 @@ public void loadLibraryRelative(String name) {
             usrPaths = tokens;
         }
         String libname = System.mapLibraryName(name);
-        if (sysPath != null && loadLibrary0(new File(sysPath, libname), false)) {
+        if (sysPath != null && loadLibraryFromPath(sysPath, libname)) {
             return;
         }
         for (String path : usrPaths) {
-            File libpath = new File(path, libname);
-            if (loadLibrary0(libpath, false)) {
-                return;
-            }
-            File altpath = Target_jdk_internal_loader_ClassLoaderHelper.mapAlternativeName(libpath);
-            if (altpath != null && loadLibrary0(altpath, false)) {
+            if (loadLibraryFromPath(path, libname)) {
                 return;
             }
         }
         throw new UnsatisfiedLinkError("no " + name + " in java.library.path");
     }
 
+    private boolean loadLibraryFromPath(String path, String libname) {
+        File libpath = new File(path, libname);
+        if (loadLibrary0(libpath, false)) {
```

Comment:
No need for this `if`, could return `loadLibrary0(libpath, false)`

Reply:
This is incorrect, directly returning `loadLibrary0` makes the altpath flow unreachable making the issue these changes are supposed to fix even worse.
If I may be so bold, why are you reviewing this as a non oracle employee without being specifically asked to review this? This now gives the impression someone is looking into this which will cause this to go unnoticed by people that have an actual say in what happens with this PR.
Please refrain from adding random reviews on PRs you have no affiliation with whatsoever, especially if the input you provide is wrong.